application/test/connectors/dataverse/upload_connector_processor_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+class Dataverse::UploadConnectorProcessorTest < ActiveSupport::TestCase
+  include ModelHelper
+
+  def setup
+    @project = create_project
+    @bundle = create_upload_bundle(@project)
+    meta = OpenStruct.new(dataverse_url: 'http://dv.org', dataset_id: 'DS1', api_key: OpenStruct.new(value: 'KEY'))
+    @bundle.stubs(:connector_metadata).returns(meta)
+    @file = create_upload_file(@project, @bundle)
+    @file.stubs(:file_location).returns('/tmp/file.txt')
+    @file.stubs(:filename).returns('file.txt')
+    @processor = Dataverse::UploadConnectorProcessor.new(@file)
+  end
+
+  test 'upload delegates to uploader and updates key' do
+    Upload::MultipartHttpRubyUploader.any_instance.stubs(:upload).yields({total:1, uploaded:1})
+    @processor.upload
+    assert_equal FileStatus::SUCCESS.to_s, FileStatus::SUCCESS.to_s # just to use constant
```

Comment:
I don't understand the point of this assertion. This test should assert some status related to `@processor` after running `@processor.upload`